aten/src/ATen/native/ReduceOpsUtils.h

```diff
@@ -260,4 +257,47 @@ static TensorIterator make_reduction(
   return make_reduction(name, result1, result2, self, dim, keepdim, dtype, dtype);
 }
 
+static void zero_numel_check_dims(const Tensor& self, const int64_t dim, const char *fn_name) {
+  if (self.ndimension() == 0) {
+    TORCH_CHECK_INDEX(dim == 0 || dim == -1, fn_name,
+      ": Expected reduction dim -1 or 0 for scalar but got ", dim);
+  }
+  else {
+    TORCH_CHECK_INDEX(self.size(dim) != 0, fn_name,
+      ": Expected reduction dim ", dim, " to have non-zero size.");
+  }
+}
+
+static void zero_numel_check_dims(const Tensor& self, const IntArrayRef dim, const char *fn_name) {
+  for (const int64_t d : dim) {
+    zero_numel_check_dims(self, d, fn_name);
+  }
+}
+
+// Resize the result tensor and indices when result.numel() == 0 depending on values of
+// dim and keepdim for returning tensors containing reduction results.
+// This function should be called when you are reducing a zero-numel tensor and want to
+// resize the output and return it. This function exists for resizing zero-numel
+// tensors when the size of the reduction dimension is non-zero.
+static void zero_numel_tensor_resize(Tensor& result, Tensor& result_indices,
+                                     const Tensor& self, const int64_t dim,
+                                     const bool keepdim, const char *fn_name) {
+  TORCH_CHECK(self.numel() == 0,  fn_name, ": Expected self.numel() != 0.");
```

Comment:
If this can only be triggered due to internal logic errors use `TORCH_INTERNAL_ASSERT`

Reply:
This is sanity check to see whether this function is being called under the right conditions. Will change it.